psydac/api/ast/fem.py

```diff
@@ -368,19 +404,22 @@ def __init__(self, expr, terminal_expr, spaces, mapping_space=None, tag=None, ma
             multiplicity_fields = get_multiplicity(fields, spaces.vector_space)
             is_parallel         = spaces.vector_space.parallel
             spaces              = spaces.symbolic_space
+
+            # Define the type of scalar that the code should manage
+            dtype           = spaces.codomain_type if hasattr(spaces, 'codomain_type') else 'real'
+
         else:
             raise NotImplementedError('TODO')
 
-        tests  = expand_hdiv_hcurl(tests)
-        trials = expand_hdiv_hcurl(trials)
-        fields = expand_hdiv_hcurl(fields)
+        tests                = expand_hdiv_hcurl(tests)
+        trials               = expand_hdiv_hcurl(trials)
+        fields               = expand_hdiv_hcurl(fields)
```

Comment:
Why these spaces?

Reply:
All the spaces are here to align the equals signs. Thus the code is more understandable